core/src/main/java/org/opentosca/toscana/core/plugin/AbstractPlugin.java

```diff
@@ -1,56 +1,48 @@
 package org.opentosca.toscana.core.plugin;
 
 import org.opentosca.toscana.core.transformation.platform.Platform;
-import org.opentosca.toscana.core.transformation.properties.Property;
+
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.util.HashSet;
-
 /**
- * This class describes a Abstract plugin and implements some things that you would have to implement
- * manually when using the plugin interface
+ * This class describes a Abstract plugin and implements some things that you would have to implement manually when
+ * using the plugin interface
  */
 public abstract class AbstractPlugin implements TransformationPlugin {
 
     /**
      * This logging object can be used for non transformation specific logging within the subclasses
      */
     protected Logger logger = LoggerFactory.getLogger(getClass());
+    private final Platform platform;
 
-    public AbstractPlugin() {
+    public AbstractPlugin(Platform platform) {
+        this.platform = platform;
+        if (platform == null) {
+            throw new IllegalStateException("initPlatform() must return the supported Platform of the plugin.");
+        }
         this.init();
-        logger.info("Initialized '{}' Plugin.", getName());
+        logger.info("Initialized '{}' Plugin.", platform.name);
     }
 
     /**
-     * @return the name of the plugins (All chars allowed)
-     */
-    public abstract String getName();
-
-    /**
-     * @return the identifier for the plugin.
-     */
-    public abstract String getIdentifier();
-
-    /**
-     * @return The plugin specific properties like enpoints get provided by this mehtod
+     * @return the supported platform of the plugin
      */
-    public HashSet<Property> getPluginSpecificProperties() {
-        return new HashSet<>();
+    protected Platform initPlatform() {
```

Comment:
This is not needed anymore when we require the insertion of the platform object into the constructor.